SIRF build with gnu 4.8.5

Using gnu 4.8.5 I have a couple of build errors:

```
`src/Registration/NiftyMoMo/BSplineTransformation.cpp: In member function ‘void NiftyMoMo::BSplineTransformation::CPGToDVF3D(nifti_image*)’:

src/Registration/NiftyMoMo/BSplineTransformation.cpp:532:76: error: ‘NiftyMoMo::BSplineTransformation::controlPointGridImage’ is not a variable in clause ‘share ’
  firstZCPG, firstYCPG, firstXCPG, indCPG, indDVF, indTmp, valX, valY, valZ)
```
This is fixed by removing controlPointGridImage from the OpenMP shared declaration in CPGToDVF2D and CPGToDVF3D which shouldn't change any functionality as anything declared before ```#pragma openmp parallel``` will be shared default.

src/common/include/sirf/common/ANumRef.h
'invalid_argument’ is not a member of ‘std’

This is fixed by including the stdexcept header

--- a/src/Registration/NiftyMoMo/BSplineTransformation.cpp
+++ b/src/Registration/NiftyMoMo/BSplineTransformation.cpp
@@ -379,7 +379,7 @@ void BSplineTransformation::CPGToDVF2D( nifti_image *denseVectorFieldImage )
   //use openmp if available and not debug build
 #if defined (NDEBUG) && defined (_OPENMP) && _OPENMP>201012
 #pragma omp parallel for default(none) \
-	shared(controlPointGridImage, denseVectorFieldImage, DVFSpacingInCPG, \
+	shared(denseVectorFieldImage, DVFSpacingInCPG, \
 	DVFOriginInCPG, CPGPtrX, CPGPtrY, DVFPtrX, DVFPtrY) \
 	private(yDVF, yCPG, xDVF, xCPG, tmpBasis, bSplineBasis, tmpXValsX, tmpXValsY, \
 	firstYCPG, firstXCPG, indCPG, indDVF, valX, valY)
@@ -525,7 +525,7 @@ void BSplineTransformation::CPGToDVF3D( nifti_image *denseVectorFieldImage )
   //use openmp if available and not debug build
 #if defined (NDEBUG) && defined (_OPENMP) && _OPENMP>201012
 #pragma omp parallel for default(none) \
-	shared(controlPointGridImage, denseVectorFieldImage, DVFSpacingInCPG, \
+	shared(denseVectorFieldImage, DVFSpacingInCPG, \
 	DVFOriginInCPG, CPGPtrX, CPGPtrY, CPGPtrZ, DVFPtrX, DVFPtrY, DVFPtrZ) \
 	private(zDVF, zCPG, yDVF, yCPG, xDVF, xCPG, tmpBasis, bSplineBasis, \
 	tmpXYValsX, tmpXYValsY, tmpXYValsZ, tmpXValsX, tmpXValsY, tmpXValsZ, \

--- a/src/common/include/sirf/common/ANumRef.h
+++ b/src/common/include/sirf/common/ANumRef.h
@@ -27,7 +27,7 @@ limitations under the License.
 
 #include <complex>
 #include <typeinfo>
-
+#include <stdexcept>
 /// Number type. Taken from ismrmrd/ismrmrd.h (saves having to include 
 /// it in this file, which should be independent of it)
 /// class NumberType

